src/main/java/gregtech/api/metatileentity/multiblock/RecipeMapMultiblockController.java

```diff
@@ -129,6 +139,11 @@ protected void initializeAbilities() {
         inputEnergy.addAll(getAbilities(MultiblockAbility.SUBSTATION_INPUT_ENERGY));
         inputEnergy.addAll(getAbilities(MultiblockAbility.INPUT_LASER));
         this.energyContainer = new EnergyContainerList(inputEnergy);
+
+        this.refreshBeforeConsumptions = getMultiblockParts().stream()
+                .filter(p -> p instanceof IRefreshBeforeConsumption)
+                .map(p -> (IRefreshBeforeConsumption) p)
+                .collect(Collectors.toList());
```

Comment:
Use a `new ArrayList<>()` and add the refreshable parts in a for loop afterwards. Streams in recipe logic are heavily discouraged for performance reasons (thanks Java 8).

Reply:
Fixed in new commit.